app/src/src/components/modals/OptionModal.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Text, View, TouchableHighlight } from 'react-native';
+import { Picker } from '@react-native-picker/picker';
+
+// common selection inside modals
+const OptionModal = ({
+  visible,
+  setVisible,
+  title,
+  type,
+  setType,
+  choices,
+  setZero,
+  styles,
+}) => {
+  const onPickerChange = (itemValue, _itemIndex) => {
+    let zero = 999;
+    if (itemValue == 0) zero = 0;
+    setType(itemValue);
+    setZero(zero);
```

Comment:
We can probably reduce line 17, 18, 20, down to one line `setZero(itemValue === 0 ? 0 : 999);`